src/btchip_helpers.c

```diff
@@ -480,19 +426,57 @@ void btchip_transaction_add_output(unsigned char *hash160Address,
 }
 
 
-void btchip_sign_finalhash(void *keyContext,
-                                 unsigned char *in, unsigned short inlen,
-                                 unsigned char *out, unsigned short outlen,
+int btchip_sign_finalhash(unsigned char* path, size_t path_len, unsigned char *in, unsigned short inlen,
+                                 unsigned char *out, size_t* outlen,
                                  unsigned char rfc6979) {
-    io_seproxyhal_io_heartbeat();
 
     unsigned int info = 0;
-    cx_ecdsa_sign((cx_ecfp_private_key_t *)keyContext,
-                    CX_LAST | (rfc6979 ? CX_RND_RFC6979 : CX_RND_TRNG),
-                    CX_SHA256, in, inlen, out, outlen, &info);
-    if (info & CX_ECCINFO_PARITY_ODD) {
-        out[0] |= 0x01;
+
+    io_seproxyhal_io_heartbeat();
+    
+    bip32_path_t bip32Path;
+    bip32Path.length = path[0];
+
+    if (!parse_serialized_path(&bip32Path, path, path_len)) {
+        return -1;
+    }
+
+    if (bip32_derive_ecdsa_sign_hash_256(
+            BTCHIP_CURVE,
+            bip32Path.path, 
+            bip32Path.length,
+            CX_LAST | (rfc6979 ? CX_RND_RFC6979 : CX_RND_TRNG), 
+            CX_SHA256,
+            in, 
+            inlen, 
+            out, 
+            outlen,
+            &info) != CX_OK) {
+        return -1;
     }
 
     io_seproxyhal_io_heartbeat();
+    return 0;
+}
+
+int btchip_get_public_key(unsigned char* keyPath, size_t keyPath_len, cx_ecfp_public_key_t* public_key, unsigned char* chainCode) {
```

Comment:
```suggestion
int btchip_get_public_key(unsigned char* keyPath, size_t keyPath_len, uint8_t raw_pubkey[static 65], unsigned char* chainCode) {
```
?
As later this is only what's used?